src/backend/distributed/commands/utility_hook.c

```diff
@@ -607,12 +774,33 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 static void
 ExecuteDistributedDDLJob(DDLJob *ddlJob)
 {
-	bool shouldSyncMetadata = ShouldSyncTableMetadata(ddlJob->targetRelationId);
+	bool shouldSyncMetadata = false;
 
 	EnsureCoordinator();
-	EnsurePartitionTableNotReplicated(ddlJob->targetRelationId);
 
-	if (!ddlJob->concurrentIndexCmd)
+	if (ddlJob->targetRelationId != InvalidOid)
+	{
+		/*
+		 * Only for ddlJobs that are targetting a relation (table) we want to sync its
+		 * metadata and verify some properties around the table.
+		 */
+		shouldSyncMetadata = ShouldSyncTableMetadata(ddlJob->targetRelationId);
+		EnsurePartitionTableNotReplicated(ddlJob->targetRelationId);
+	}
+
+	if (TaskExecutorType != MULTI_EXECUTOR_ADAPTIVE &&
+		ddlJob->targetRelationId == InvalidOid)
+	{
+		/*
+		 * Some ddl jobs can only be run by the adaptive executor and not our legacy ones.
+		 *
+		 * These are tasks that are not pinned to any relation nor shards. We can execute
+		 * these very naively with a simple for loop that sends them to the target worker.
+		 */
+
+		ExecuteNodeBaseDDLCommands(ddlJob->taskList);
```

Comment:
Our `check-multi` test suite is still running on the old executors. Which is a good thing since we have not removed them.

Without implementing this it would cause a big difference in output files which I like less then supporting it like this.